### PR TITLE
Add PyTorch fork with NestedTensor support as submodule + softmax via _foreach_softmax

### PR DESCRIPTION
**Pull request by cpuhrsch**
This PR introduces PyTorch as a branch for the source of development and binary deployment for NestedTensor. To exercise this it depends on a fork of pytorch that has a branch with specific new features that otherwise don't exist (_foreach_softmax). That means, if this setup doesn't work, this code shouldn't compile nor pass CI.

This PR also cuts down the CI to cpu-only tests and removes all nightly builds. We'll then reintroduce those features incrementally as CI support for submodule builds improves. This project is meant to still work without a submodule build.

Follow-up items:
- Bring back CUDA unit tests based on submodule
- Introduce builds that don't use this submodule via extra build flag
- Bring back nightlies based on submodule

Addresses https://github.com/pytorch/nestedtensor/issues/240

### CODE REVIEW

**Comment by gchanan** [comment]
what happens with the version?  How do you distinguish between your modified version of pytorch and the normal one?
